VideoLocker/src/test/java/org/edx/mobile/test/BaseTestCase.java

```diff
@@ -18,13 +35,64 @@
 
     protected final Logger logger = new Logger(getClass().getName());
 
+    protected Context context;
+    protected Config config;
+    protected CustomGuiceModule module;
     @Before
     public void setUp() throws Exception {
-        Environment environment = new Environment();
-        environment.setupEnvironment(RuntimeEnvironment.application);
+        context = RuntimeEnvironment.application;
+        config = createConfig();
+        // Set up a new config instance that serves the mock host url
+        JsonObject properties;
+        try {
+            InputStream in = context.getAssets().open("config/config.json");
+            JsonParser parser = new JsonParser();
+            JsonElement config = parser.parse(new InputStreamReader(in));
+            properties = config.getAsJsonObject();
+        } catch (Exception e) {
+            properties = new JsonObject();
+            logger.error(e);
+        }
+
+        config = new Config(properties);
+
+        module = new CustomGuiceModule();
         print("Started Test Case: " + getClass().getName());
     }
 
+    protected void glueInjections(){
+        addBindings();
+        if ( !module.isEmpty()  ) {
+            Injector injector = RoboGuice.getOrCreateBaseApplicationInjector(RuntimeEnvironment.application, RoboGuice.DEFAULT_STAGE,
+                (Module) RoboGuice.newDefaultRoboModule(RuntimeEnvironment.application), module);
+            inject(injector);
+        }
+    }
+
+    /**
+     * subclass should inject the properties marked as @Inject
+     */
+    protected void inject(Injector injector ){}
+
+    protected void addBindings(){}
+
+    protected Config createConfig(){
+        // Set up a new config instance that serves the mock host url
+        JsonObject properties;
+        try {
+            InputStream in = context.getAssets().open("config/config.json");
+            JsonParser parser = new JsonParser();
+            JsonElement config = parser.parse(new InputStreamReader(in));
+            properties = config.getAsJsonObject();
+        } catch (Exception e) {
+            properties = new JsonObject();
+            logger.error(e);
+        }
+        return new Config(properties);
```

Comment:
Moving config creating to the base test class doesn't make sense, as the mock server isn't initialized at this point. It is overridden in the `HttpBaseTestCase` subclass, but at that point it's `setUp()` is not called and server is not initialized so it throws a `NullPointerException`. Also, since the global `Config` has been ported to a Roboguice singleton, we need a way to actually replace it with a custom test URL.